pysamstats/io.py

```diff
@@ -149,6 +149,9 @@ def write_hdf5(stats_type, outfile, alignmentfile, fields=None, progress=None, h
     # determine dtype
     if dtype is None:
         dtype = dict(default_dtype)
+        max_seqid_len = determine_max_seqid(alignmentfile)
+        dtype["chrom"] = "a{0}".format(max_seqid_len)
+
     else:
         default_dtype = dict(default_dtype)
         default_dtype.update(dict(dtype))
```

Comment:
I think as structured this means the dtype will only be set according to the maximum chromosome name length if the user provides no dtype. But the user could provide a partial dtype not including chrom field. Think this needs to be something like:

```python
default_dtype = dict(default_dtype)
max_seqid_len = determine_max_seqid(alignmentfile)
default_dtype["chrom"] = "a{0}".format(max_seqid_len)
if dtype is not None:
    default_dtype.update(dict(dtype))
dtype = default_dtype
```